tests/PHPUnit/Unit/DataAccess/LogQueryBuilder/JoinGeneratorTest.php

```diff
@@ -102,6 +102,34 @@ public function test_generate_getJoinString_manuallyJoinedAlready()
         $this->assertEquals($expected, $generator->getJoinString());
     }
 
+    public function test_generate_getJoinString_manuallyJoinedAlreadyWithCustomConditionInArray()
+    {
+        $generator = $this->generate(array(
+            'log_visit',
+            array('table' => 'log_conversion', 'joinOn' => 'log_visit.idvisit2 = log_conversion.idvisit2'),
+            'log_conversion'
+        ));
```

Comment:
Haven't had a closer look at the usage of the `JoinGenerator`, but the result might still be invalid if the order changes like this
```
        $generator = $this->generate(array(
            'log_visit',
            'log_conversion',
            array('table' => 'log_conversion', 'joinOn' => 'log_visit.idvisit2 = log_conversion.idvisit2')
        ));
```
That would still result in a duplicate `LEFT JOIN`. Is that something that could happen, or can we ignore that?